### ISSUE TITLE
Azure Storage Explore cannot parse filelist if metadata's key is _

### ISSUE THREAD

**Issue by zelon** [microsoft/AzureStorageExplorer#2954]
**Storage Explorer Version**: 1.13.0
**Build Number**: 20200430.5
**Platform/OS**: Windows 10
**Architecture**: x64
**Regression From**:  I don't exaly version. But previous version was working fine

## Bug Description ##
When I reach a blob container. below error message show up and Azure Storage Explorer stop working.
"Error "TypeError: obj[charkey].match is not a function" occurred while parsing the response body....."

## Steps to Reproduce ##
Just launch Microsoft Azure Storage Explorer and access my blob container

## Expected Experience ##
Show my files on the blob container

## Actual Experience ##
Show "Error "TypeError: obj[charkey].match is not a function" occurred while parsing the response body....." and stop working

## Additional Context ##
After some test cases, I found that Azure Storage Explorer cannot parse a file that contains metadata with '_' as key. For example, if I add "key:_, value:test", Azure Storage Explore show the error message

**Comment by MRayermannMSFT** [comment]
@zelon are you saying that you have a metadata whose key is an empty string? From what I can see that shouldn't be possible. 🤨

Here's the error I get from the server if I force us to send a metadata whose key is empty string:
```
<Error><Code>EmptyMetadataKey</Code><Message>The key for one of the metadata key-value pairs is empty.\nRequestId:f1096162-001e-00c6-3652-3555c1000000\nTime:2020-05-29T00:45:50.5959621Z</Message>
```

As for key of _, not only can I repro this, I now cannot access some of my containers!!! 😂 Digging through the code, this looks to be coming from the Azure Blob SDK. I'll work on reporting this to them.

**Comment by MRayermannMSFT** [comment]
@zelon I've opened a bug here if you'd like to track it: https://github.com/Azure/azure-sdk-for-js/issues/9171

**Comment by MRayermannMSFT** [comment]
I don't think SDK team is going to get to this in time for 1.15, so moving to 1.16. If they are able to get a fix in though then I'll gladly move this back.

**Comment by MRayermannMSFT** [comment]
Still no updates from the SDK team. Moving this to future.

**Comment by jeremymeng** [comment]
@zelon we are still testing a fix to https://github.com/Azure/azure-sdk-for-js/issues/9171.  Meanwhile, is the workaround of not using underscore as metadata key good enough for you?

**Comment by zelon** [comment]
I could access via portal.azure.com and remove the key

**Comment by MRayermannMSFT** [comment]
SDK has updated! Moving this to 1.19 to remind us to do an SDK update for that milestone.

**Comment by MRayermannMSFT** [comment]
This is now definitely going to be fixed as of 1.19. It might even be fixed in 1.18? I'd have to go back and look, but for now, I am closing this since I know we have a new enough SDK as of 1.19. :)
